lib/index.js

```diff
@@ -34,137 +31,161 @@ module.exports = (opts, ctx) => {
-        ctx.siteConfig.title = name
-      }
-      if (!ctx.siteConfig.description) {
-        ctx.siteConfig.description = description
-      }
+        spinner.start(`Fetching TOC ... `)
 
-      let defaultActionLink
+        const toc = await yuque.getToc()
+        assert(toc && toc.data, `[${PACKAGE_NAME}] cannot get toc of ${repoId}`)
 
-      // Apply pages
-      for (const page of toc.data) {
-        const { title, slug } = page
-        // Once the slug is equal to '#', it means that current node
-        // is an empty node.
-        if (!slug || slug === '#') {
-          continue
-        }
+        debug('toc', toc.data)
 
-        spinner.start(`Fetching ${chalk.cyan(title)} ... `)
+        spinner.succeed(`Retrieved TOC`)
 
-        const { status, data } = await yuque.getPage(slug)
-        debug('status = ', status)
+        // Apply sidebar config
+        ctx.siteConfig.themeConfig = ctx.siteConfig.themeConfig || {}
+        if (!ctx.siteConfig.title) {
+          ctx.siteConfig.title = name
+        }
+        if (!ctx.siteConfig.description) {
+          ctx.siteConfig.description = description
+        }
 
-        let postContent
-        if (status && status === 404) {
-          postContent = `# ${title}\n > 此文档尚未创建`
+        const directory = getSidebarByToc(toc.data, repoRoute && `/${repoRoute}`)
+        if (!ctx.siteConfig.themeConfig.sidebar) {
+          ctx.siteConfig.themeConfig.sidebar = {}
+        }
+
+        if (repoRoute) {
+          ctx.siteConfig.themeConfig.sidebar[`/${repoRoute}/`] = directory
         } else {
-          const useMarkdown = typeof source === 'string'
-            ? source === 'markdown'
-            : typeof source === 'function'
-              ? source(page) === 'markdown'
-              : false
-          if (useMarkdown || data.format === 'markdown') {
-            postContent = data.body || ''
+          ctx.siteConfig.themeConfig.sidebar = directory
+        }
+        let defaultActionLink
+
+        // Apply pages
+        for (const page of toc.data) {
+          const { title, slug } = page
+          // Once the slug is equal to '#', it means that current node
+          // is an empty node.
+          if (!slug || slug === '#') {
+            continue
+          }
+
+          spinner.start(`Fetching ${chalk.cyan(title)} ... `)
+
+          const { status, data } = await yuque.getPage(slug)
+          debug('status = ', status)
+
+          let postContent
+          if (status && status === 404) {
+            postContent = `# ${title}\n > 此文档尚未创建`
           } else {
-            postContent = prettify(data.body_html || '')
+            const useMarkdown =
+              typeof source === 'string'
+                ? source === 'markdown'
+                : typeof source === 'function'
+                ? source(page) === 'markdown'
+                : false
+            if (useMarkdown || data.format === 'markdown' || data.format === 'lake') {
```

Comment:
I came across some docs where the format is lake, and it would parse data.body_html instead of data.body. In this case the build command will emit compile error and fail , so I have to treat lake format as markdown to get around it. I don't know the initial motive of this design, so I think I need some opinion on whether this change would break anything.

Reply:
If you parse the "lake" as markdown by default, you'll lose some of the styles which is only available in `body_html`, e.g. width of images.